### PR TITLE
Implemenet k-vector computation

### PR DESCRIPTION
**Pull request by PicoCentauri**
Added computation of k vectors for LODE calculations. I will add the tests but if @Luthaf or @kvhuguenin already have ideas please let me know.

### CODE REVIEW

File: rascaline/src/math/kvectors.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    /// 3 component k-vector
+    pub vector: Vector3D,
+    /// length of the k-vector
+    pub norm: f64,
+}
+
+pub fn compute_kvectors(cell: &Matrix3, kcutoff: f64) -> Vec<KVector> {
+    // Generate k-vectors up to a certain cutoff (in reciprocal space units) 
+    // for a given cell.
+    let reciprocal_cell = 2.0 * std::f64::consts::PI * cell.transposed().inverse();
```
Comment (Luthaf) [inline rascaline/src/math/kvectors.rs:19]: You should add a sanity check that the cell is defined, something like 

```suggestion
/// Generate k-vectors up to a certain cutoff (in reciprocal space units) 
/// for a given cell.
pub fn compute_kvectors(cell: &Matrix3, kcutoff: f64) -> Vec<KVector> {
    debug_assert_ne!(cell, Matrix3::zero(), "invalid cell matrix");

    let reciprocal_cell = 2.0 * std::f64::consts::PI * cell.transposed().inverse();
```

Or take a `UnitCell` as input and check the shape here!

Comment (PicoCentauri) [reply rascaline/src/math/kvectors.rs:19]: I like the approach with the matrix, because this is really general. Thanks for the sanity checks!

**Comment by Luthaf** [comment]
Looks good overall, could you rebase and squash before we merge? (Or I'll use a squash merge on github, as you prefer!)

**Comment by PicoCentauri** [comment]
Nice, we can squash merge here.

**Comment by Luthaf** [comment]
Sorry, I missed your message here

**Comment by PicoCentauri** [comment]
No worries thanks for the merge.

**Pull request merged**